marker_api/src/ast/ty/fn_ty.rs

```diff
@@ -30,26 +28,6 @@ impl<'ast> SemFnTy<'ast> {
     }
 }
 
-/// The syntactic representation of a
-/// [closure type](https://doc.rust-lang.org/reference/types/closure.html).
```

Comment:
That's weird how there could be a syntactic representation of a closure type because there is no syntax to denote a type of the closure except for `impl Fn*`. However,  `impl Fn*` should not be coupled with closures, that's a broader concept.

So I wonder what this struct represented then

Reply:
I don't actually know anymore why I added this struct. My guess is that this is a left over from an old architecture. There was a time, where I planned to merge both syntactic and semantic types into a single representation. My idea was that this would make it easier, to work with types. Then you could compare syntactic and semantic types etc.

However, while working on semantic types, I quickly noticed that these are way more different than I thought. It also became clear that this combination will most likely be painful for everyone involved, with very few benefits.

During the move to split semantic and syntactic types, I just added the `Syn` prefix to most types. The struct might have been intended to be a semantic type, but got caught in the crossfire.